component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPTokenGenerator.java

```diff
@@ -167,6 +170,52 @@ public static String generateTOTPTokenLocal(String username, AuthenticationConte
 		return Long.toString(token);
 	}
 
+	/**
+	 * Generate 6 digit TOTP token for a given secret key and time index.
+	 *
+	 * @param secret  Secret key in binary format.
+	 * @param context Authentication context.
+	 * @return Six digit TOTP token value as a long.
+	 * @throws NoSuchAlgorithmException If the specific algorithm was not found.
+	 * @throws InvalidKeyException      If an invalid signKey provided.
+	 * @throws TOTPException            If an error occurred while getting the time index.
+	 */
+	private static long generateToken(byte[] secret, AuthenticationContext context)
+			throws NoSuchAlgorithmException, InvalidKeyException, TOTPException {
+
+		long token = getCode(secret, getTimeIndex(context));
+		// We need to check whether the token at least have the minimum number of digits.
+		if (isTokenHasMinimumChars(token)) {
```

Comment:
Let's add some debug logs with the token, so it would easy to identify the length.